tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,328 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data Processing for NeMo 2.0 LLMs\n",
+    "\n",
+    "This tutorial will cover the steps to go from a raw pretraining dataset all the way to configuring the data module for pretraining using a NeMo 2.0 recipe.\n",
+    "We will be using the [SlimPajama-627B](https://huggingface.co/datasets/cerebras/SlimPajama-627B>) dataset for reference. We will also show how to exclude certain sources from the dataset, for instance, by default we will be excluding all data from the `RedPajamaBook` set.\n",
+    "\n",
+    "This tutorial involves four steps:\n",
+    "\n",
+    "1. Downloading data\n",
+    "2. Extracting data\n",
+    "3. Concatenating data\n",
+    "4. Preprocessing data for NeMo 2.0/Megatron\n",
+    "\n",
+    "First, we'll define each step. Next, we will see how we can use NeMo-Run to execute the steps sequentially on your local workstation using Docker or on Slurm.\n",
+    "\n",
+    "### Pre-requisites\n",
+    "This notebook assumes familiarity with [NeMo-Run](https://github.com/NVIDIA/NeMo-Run). Additionally, the docker execution and slurm execution steps require access to docker on your host and a remote slurm cluster respectively.\n",
+    "Additionally, you will have to complete the following steps:\n",
+    "\n",
+    "1. Set HOST_DATA_PATH in the first cell to a parent folder on your workstation where you want to save the data.\n",
+    "1. Create directories `HOST_DATA_PATH/tokenizer` and `HOST_DATA_PATH/slimpajama`.\n",
+    "1. Download the Llama 3 `tokenizer.model` file either from [Huggingface](https://huggingface.co/meta-llama/Llama-2-7b/blob/main/tokenizer.model) or https://www.llama.com/llama-downloads/ and place it at `{HOST_DATA_PATH}/tokenizer/tokenizer.model`.\n",
+    "    For HF, you can do it by running \n",
+    "    ```bash\n",
+    "    HF_TOKEN=... huggingface-cli download meta-llama/Llama-2-7B tokenizer.model --local-dir {HOST_DATA_PATH}/tokenizer/\n",
+    "    ```\n",
+    "\n",
+    "> [!NOTE]\n",
+    "> All code for this tutorial can be found at https://github.com/NVIDIA/NeMo/tree/main/examples/llm/slimpajama."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nemo_run as run\n",
+    "\n",
+    "from data.download import download_slimpajama\n",
+    "from data.extract import run_extraction\n",
+    "from data.preprocess import preprocess_data\n",
+    "\n",
+    "HOST_DATA_PATH = \"/data\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Downloading Data\n",
+    "\n",
+    "First, we will configure the task to download data from Huggingface. We will use the Huggingface CLI for this. The function that configures the download script can be found [here](./data/download.py)"
```

Comment:
fix name and capitalization

First, we will configure the task to download data from Hugging Face. We will use the Hugging Face CLI for this. The function that configures the download script can be found [here](./data/download.py)"